File: Python/Ram.py
from Komponen import Komponen

class Ram(Komponen):
    def __init__(self, merk, nama, kapasitas_gb):
        super().__init__(merk, nama)
        self.__kapasitas_gb = kapasitas_gb

    # Getter
    def get_kapasitas_gb(self):
        return self.__kapasitas_gb

    # Setter
    def set_kapasitas_gb(self, kapasitas_gb):
        self.__kapasitas_gb = kapasitas_gb

File: Cpp/Ram.cpp
#pragma once
#include <iostream>
#include <string>
#include "Komponen.cpp"

using namespace std;

class Ram : public Komponen {
private:
    int kapasitasGB;
public:
    Ram() {}

    Ram(int kapasitasGB, string merk, string nama) 
        : Komponen(merk, nama) {
        this->kapasitasGB = kapasitasGB;
    }

    void setKapasitasGB(int kapasitasGB) {
        this->kapasitasGB = kapasitasGB;
    }

    int getKapasitasGB() const { 
        return this->kapasitasGB;
    }
    

    ~Ram() {}
};
